android/src/main/cpp/LayoutAnimations.cpp

```diff
@@ -17,70 +17,50 @@ void LayoutAnimations::setWeakUIRuntime(std::weak_ptr<jsi::Runtime> wrt) {
     alias_ref<JMap<jstring, jstring>> values) {
-  if (auto rt = this->weakUIRuntime.lock()) {
-    jsi::Value layoutAnimationRepositoryAsValue =
-        rt->global()
-            .getPropertyAsObject(*rt, "global")
-            .getProperty(*rt, "LayoutAnimationRepository");
-    if (!layoutAnimationRepositoryAsValue.isUndefined()) {
-      jsi::Function startAnimationForTag =
-          layoutAnimationRepositoryAsValue.getObject(*rt).getPropertyAsFunction(
-              *rt, "startAnimationForTag");
-      jsi::Object target(*rt);
-
-      for (const auto &entry : *values) {
-        target.setProperty(
-            *rt,
-            entry.first->toStdString().c_str(),
-            std::stof(entry.second->toStdString()));
-      }
-      startAnimationForTag.call(
-          *rt,
-          jsi::Value(tag),
-          jsi::String::createFromUtf8(*rt, type->toStdString()),
-          target);
-    }
-  }
+  this->animationStartingBlock_(tag, type, values);
 }
 
-void LayoutAnimations::removeConfigForTag(int tag) {
-  if (auto rt = this->weakUIRuntime.lock()) {
-    jsi::Value layoutAnimationRepositoryAsValue =
-        rt->global()
-            .getPropertyAsObject(*rt, "global")
-            .getProperty(*rt, "LayoutAnimationRepository");
-    if (!layoutAnimationRepositoryAsValue.isUndefined()) {
-      jsi::Function removeConfig =
-          layoutAnimationRepositoryAsValue.getObject(*rt).getPropertyAsFunction(
-              *rt, "removeConfig");
-      removeConfig.call(*rt, jsi::Value(tag));
-    }
-  }
-}
-
-void LayoutAnimations::notifyAboutProgress(
-    const jsi::Value &progress,
-    int tag) {
+void LayoutAnimations::progressLayoutAnimation(
+    int tag,
+    const jsi::Value &progress) {
   if (auto rt = this->weakUIRuntime.lock()) {
     static const auto method =
         javaPart_->getClass()
-            ->getMethod<void(JMap<JString, JObject>::javaobject, int)>(
-                "notifyAboutProgress");
+            ->getMethod<void(int, JMap<JString, JObject>::javaobject)>(
+                "progressLayoutAnimation");
     method(
         javaPart_.get(),
-        JNIHelper::ConvertToPropsMap(*rt, progress.asObject(*rt)).get(),
-        tag);
+        tag,
+        JNIHelper::ConvertToPropsMap(*rt, progress.asObject(*rt)).get());
   }
 }
 
-void LayoutAnimations::notifyAboutEnd(int tag, int cancelled) {
+void LayoutAnimations::endLayoutAnimation(
+    int tag,
+    bool cancelled,
+    bool removeView) {
   static const auto method =
-      javaPart_->getClass()->getMethod<void(int, int)>("notifyAboutEnd");
-  method(javaPart_.get(), tag, cancelled);
+      javaPart_->getClass()->getMethod<void(int, bool, bool)>(
+          "endLayoutAnimation");
+  method(javaPart_.get(), tag, cancelled, removeView);
+}
+
+void LayoutAnimations::setHasAnimationBlock(
+    HasAnimationBlock hasAnimationBlock) {
+  this->hasAnimationBlock_ = hasAnimationBlock;
+}
+
+bool LayoutAnimations::hasAnimationForTag(int tag, std::string type) {
```

Comment:
```suggestion
bool LayoutAnimations::hasAnimationForTag(int tag, const std::string &type) {
```

Reply:
This doesn't compile - this function is called from Java, and it seems `MakeNativeMethod` doesn't like functions which take references.